students/chris_lombardi/Lesson03/assignment/basic_operations.py

```diff
@@ -0,0 +1,97 @@
+"""Module for unit testing the methods of the basic_operations and customer_model modules"""
+
+#pylint: disable=unused-wildcard-import
+#pylint: disable=wildcard-import
+#pylint: disable=too-many-arguments
+
+import logging
+from customer_model import *
+
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+def add_customer(customer_id, name, lastname, home_address, phone_number,
+                 email_address, status, credit_limit):
+    """
+    Add a new customer to the customer database using the provided params.
+    """
+
+#    CUST_SCHEMA = {'cust_id': 0, 'first_name': 1, 'last_name': 2, 'address': 3,
+#                   'city': 4, 'state': 5, 'zip': 6, 'phone_number': 7, 'email': 8,
+#                   'status': 9, 'credit_limit': 10}
+    try:
+        with database.transaction():
+            logging.info('Trying to add customer %s...', customer_id)
+            new_cust = Customer.create(
+                cust_id=customer_id,
+                cust_firstname=name,
+                cust_lastname=lastname,
+                cust_address=home_address,
+                cust_phone=phone_number,
+                cust_email=email_address,
+                cust_status=status,
+                cust_credit_limit=credit_limit
+                )
+            new_cust.save()
+            LOGGER.info('Added new customer %s %s', name, lastname)
+
+    except IntegrityError as error_1:
+        LOGGER.info(error_1)
+        LOGGER.info('Non-unique customer id: %s', customer_id)
+    except Exception as error_2:
```

Comment:
Avoid using _catch all_ exception handling. The risk for released code is that you could eventually run into a corner case that trigger a different exception, which is then handled (incorrectly) by the catch all.